### PR TITLE
add redux to quest page and quests list and rename guestsList to Ques…

### PR DESCRIPTION
**Pull request by lalka-anka**
rename guests... to quests... and add redux to quest page and quests list, but there are still components with simple fetching that also need to be fixed

### CODE REVIEW

File: src/components/QuestPage/QuestDescriptionLogic.jsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ class GuestDescription extends Component {
                         fill="black"/>
                 </svg>;
                 description = <ReactMarkdown source={this.props.quest.description}/>;
-                results = 'привет';
+                results = '';
```
Comment (lalka-anka) [inline src/components/QuestPage/QuestDescriptionLogic.jsx:92]: это я чекала модуль))))))

File: src/components/Account/AccountTemplate/QuestsList.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import {Link} from "react-router-dom";
+import './QuestsList.css'
+import { fetchQuestsListInfo } from '../../../redux/Actions/QuestsListApi'
+import {connect} from "react-redux";
+
+
+class QuestsList extends Component {
+
+  componentDidMount () {
+    this.props.fetchQuestsListFromRedux()
```
Comment (toplenboren) [inline src/components/Account/AccountTemplate/QuestsList.jsx:14]: If there are only redux fetching let's rename it to `fetchQuestList` ?

File: src/components/Account/AccountTemplate/QuestsList.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  mapQuestsToTemplate () {
+      return this.props.questsListFromRedux.map((obj) =>
+          <Col key={'quest:' + obj.id} xs={22} md={8}>
+            <Link to={'/quests/' + obj.id}><QuestCard quest={obj} isInfinite={obj.isInfinite}/></Link>
+          </Col>
+      )
+  }
+
+  getRepresentationByState () {
+    if (this.props.questsListFromReduxIsFetching)
```
Comment (toplenboren) [inline src/components/Account/AccountTemplate/QuestsList.jsx:26]: `this.props.questsList` <- Let's make it like that :)

File: src/components/Account/AccountTemplate/QuestsList.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  questsListFromRedux: store.questsListReducer.quests,
+  questsListFromReduxIsFetching: store.questsListReducer.isFetching
+});
+
+const mapDispatchToProps = dispatch => ({
+  fetchQuestsListFromRedux: () => dispatch(fetchQuestsListInfo())
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(QuestsList)
+
```
Comment (toplenboren) [inline src/components/Account/AccountTemplate/QuestsList.jsx:61]: 61 Lines! 👍

File: src/redux/Actions/QuestsListApi.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import {
```
Comment (toplenboren) [inline src/redux/Actions/QuestsListApi.js:1]: It's better to place it to the `QuestsApi.js` than to create a separate file only for list

File: src/redux/store.js [hunk 6]
```diff
@@ -5,12 +5,14 @@ import { composeWithDevTools } from 'redux-devtools-extension';
...
 import authReducer from "./Reducers/AuthReducer";
 import questsReducer from './Reducers/QuestsReducer'
+import questsListReducer from './Reducers/QuestsListReducer'
 
 const loggerMiddleware = createLogger();
 
 const rootReducer = combineReducers({
     authReducer,
-    questsReducer
+    questsReducer,
```
Comment (toplenboren) [inline src/redux/store.js:14]: Let's make something like 

```(JavaScript)

const questsReducer = combineReducers({
  questReducer,
  questListReducer
})

const rootReducer = combineReducers({
  authReducer,
  questsReducer
})
```

Comment (toplenboren) [reply src/redux/store.js:14]: What do you think ?

File: src/redux/Reducers/QuestsListReducer.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+} from '../Actions/QuestsListActions'
+
+export default function questsReducer (
+    state = {
+        isFetching: false,
+        quests: null
+    }, action ) {
+    switch (action.type) {
+        case REQUEST_QUESTS_LIST_INFO:
+            return Object.assign({}, state, {
```
Comment (toplenboren) [inline src/redux/Reducers/QuestsListReducer.js:14]: BTW we are able to write:
```
return {...state, isFetching: true}
```

File: src/redux/Actions/QuestsListApi.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    receiveQuestsListInfo,
+    requestQuestsListInfo,
+    deleteQuestsListInfo
+} from '../Actions/QuestsListActions'
+
+import { BASE_URL } from './../../settings'
+
+export function fetchQuestsListInfo () {
+    return dispatch => {
```
Comment (toplenboren) [inline src/redux/Actions/QuestsListApi.js:10]: Please note that in #60 I've created a CommonApi.js where i leave some helper functions to fetch API with Auth header

We might need to refactor this buddy

File: src/redux/Actions/QuestsListApi.js (hunk 5 above)
Comment (lalka-anka) [reply src/redux/Actions/QuestsListApi.js:1]: but we have different fetches here: BASE_URL + '/quests/' + id and BASE_URL + '/quests') -- is it okay to merge them

**Pull request merged**